### ISSUE TITLE
TypeError: type 'typing.TypeVar' is not an acceptable base type 

### ISSUE THREAD

**Issue by Manuelrivas456** [ConservationInternational/trends.earth#806]
TypeError: type 'typing.TypeVar' is not an acceptable base type 
Traceback (most recent call last):
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 401, in loadPlugin
    __import__(packageName)
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users/Manuel Rivas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\__init__.py", line 117, in 
    from . import conf  # noqa: autoimport
    ^^^^^^^^^^^^^^^^^^
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users/Manuel Rivas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\conf.py", line 10, in 
    from te_schemas.algorithms import ExecutionScript
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\te_schemas\algorithms.py", line 6, in 
    import marshmallow_dataclass
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\marshmallow_dataclass\__init__.py", line 66, in 
    import typing_inspect
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\typing_inspect.py", line 23, in 
    from typing_extensions import _TypedDictMeta as _TypedDictMeta_TE
  File "C:\PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python\qgis\utils.py", line 892, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\typing_extensions.py", line 1174, in 
    class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
TypeError: type 'typing.TypeVar' is not an acceptable base type

Versión de Python: 3.12.3 (main, Apr 14 2024, 17:21:43) [MSC v.1938 64 bit (AMD64)] 
Versión de QGIS: 3.34.6-Prizren Prizren, 623828f5 

Ruta de Python:
C:/PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:/Users/Manuel Rivas/AppData/Roaming/QGIS/QGIS3\profiles\default/python
C:/Users/Manuel Rivas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins
C:/PROGRA~1/QGIS33~1.6/apps/qgis-ltr/./python/plugins
C:\PROGRA~1\QGIS33~1.6\apps\grass\grass83\etc\python
C:\WINDOWS\System32
C:\Program Files\QGIS 3.34.6\bin\python312.zip
C:\PROGRA~1\QGIS33~1.6\apps\Python312\DLLs
C:\PROGRA~1\QGIS33~1.6\apps\Python312\Lib
C:\Program Files\QGIS 3.34.6\bin
C:\PROGRA~1\QGIS33~1.6\apps\Python312
C:\PROGRA~1\QGIS33~1.6\apps\Python312\Lib\site-packages
C:\PROGRA~1\QGIS33~1.6\apps\Python312\Lib\site-packages\win32
C:\PROGRA~1\QGIS33~1.6\apps\Python312\Lib\site-packages\win32\lib
C:\PROGRA~1\QGIS33~1.6\apps\Python312\Lib\site-packages\Pythonwin
C:/Users/Manuel Rivas/AppData/Roaming/QGIS/QGIS3\profiles\default/python
C:\Users\Manuel Rivas\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\HCMGIS/forms

**Comment by ECuesta-M** [comment]
Hi,

I have the same issue, Did you manage to solve the problem?

**Comment by Manuelrivas456** [comment]
no todavía no funciona la aplicación

**Comment by Manuelrivas456** [comment]
![image](https://github.com/ConservationInternational/trends.earth/assets/168492872/21ddfec3-3bdc-4bb5-b3df-7247143f6a88)

**Comment by Manuelrivas456** [comment]
> ![image](https://private-user-images.githubusercontent.com/168492872/341215006-21ddfec3-3bdc-4bb5-b3df-7247143f6a88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MzU4NDMsIm5iZiI6MTcxODgzNTU0MywicGF0aCI6Ii8xNjg0OTI4NzIvMzQxMjE1MDA2LTIxZGRmZWMzLTNiZGMtNGJiNS1iM2RmLTcyNDcxNDNmNmE4OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOVQyMjE5MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMTU0MTAxYmFkYTk2ZmI2N2Y3MTQyNWJmMzQxMDViODI4MTM0OWFkZWU5OGRhMGJkZjc2Nzk3ZDU3Y2UxMDYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V4LRn8OIIIPEdRky3C2KKJfIcVoCKUfiFIEkQqC4LJs)

este es el error y todavia continua

**Comment by zirneklitis** [comment]
Hi.

I have installed the latest “typing_extensions” (4.12.2), but it did not slve the issue:
```
Couldn't load plugin 'LDMP' 

TypeError: type 'typing.TypeVar' is not an acceptable base type 
Traceback (most recent call last):
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 404, in loadPlugin
    __import__(packageName)
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/__init__.py", line 117, in 
    from . import conf  # noqa: autoimport
    ^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/conf.py", line 10, in 
    from te_schemas.algorithms import ExecutionScript
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs/te_schemas/algorithms.py", line 6, in 
    import marshmallow_dataclass
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs/marshmallow_dataclass/__init__.py", line 66, in 
    import typing_inspect
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs/typing_inspect.py", line 23, in 
    from typing_extensions import _TypedDictMeta as _TypedDictMeta_TE
  File "/usr/lib64/python3.12/site-packages/qgis/utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zirneklitis/.local/share/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs/typing_extensions.py", line 1174, in 
    class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
TypeError: type 'typing.TypeVar' is not an acceptable base type


Python versija: 3.12.3 (main, Apr 17 2024, 00:00:00) [GCC 13.2.1 20240316 (Red Hat 13.2.1-7)] 
QGIS versija: 3.38.0-Grenoble Grenoble, exported 
```

**Comment by emcasagrande** [comment]
Hi, I have the same problem with 
_Python version_: 3.12.4 (main, Jun 10 2024, 12:48:35) [MSC v.1938 64 bit (AMD64)]
_QGIS version_: 3.34.8-Prizren Prizren, 91642333

**Comment by azvoleff** [comment]
This is fixed by dff8ced73f8ba0e00ca53a807257ab04a91a22f5. A new release will be coming to the repository shortly. In the meantime the [development zipfile](https://github.com/ConservationInternational/trends.earth?tab=readme-ov-file#development-version-from-zipfile) should work.

**Comment by emcasagrande** [comment]
Dear Alex,

Thank you so much for the quick response and for sending the zip file. I
just wanted to know that now it works properly!
Have a great day.
Best,

 *Enrico CASAGRANDE,*
*Environmental Engineer, Ph.D in Agricultural Sciences*
***@***.*** ***@***.***> *
  <http://www.inra.fr/>
